lua/cfc_weapon_lockouts/server/sv_locker.lua

```diff
@@ -0,0 +1,216 @@
+            local newLength = lineLength + class:len()
+
+            if newLength <= maxLength then
+                if newLength + maxAppend:len() < maxLength then
+                    msg = msg .. class .. " "
+                    lineLength = newLength + 1
+                else
+                    msg = msg .. maxAppend
+                    break
+                end
+            else
+                msg = msg .. maxAppend
+                break
+            end
+        end
+    end
+
+    if CFCNotifications then
+        CFCNotifications.new( identifier, "Text", true )
+        local notif = CFCNotifications.get( identifier )
+
+        notif:SetTitle( "CFC WeaponLockouts" )
+        notif:SetText( msg )
+        notif:SetTextColor( color_white )
+        notif:SetDisplayTime( CFCWeaponLockouts.LOCKOUT_TIME - CFCWeaponLockouts.WARN_BUILDUP )
+        notif:SetPriority( CFCNotifications.PRIORITY_LOW )
+        notif:SetCloseable( true )
+        notif:SetIgnoreable( true )
+        notif:SetTimed( true )
+
+        notif:Send( ply )
+    else
+        ply:ChatPrint( msg )
+    end
+
+    lockWarns[ply] = {}
+end
+
+hook.Add( "PlayerSpawn", "CFC_WeaponLockouts_UnlockOnSpawn", function( ply )
+    ply.weaponLockouts = {}
+    ply.weaponLockoutWeapons = {}
+    lockWarns[ply] = {}
+
+    --Clear out warn info from weapons spawned by default
+    timer.Create( "CFC_WeaponLockouts_LockWarn_" .. ply:SteamID(), 0, 1, function()
+        lockWarns[ply] = {}
+    end )
+end )
+
+hook.Add( "PlayerDroppedWeapon", "CFC_WeaponLockouts_LockWeapon", function( ply, wep )
+    CFCWeaponLockouts.lockWeapon( ply, wep, true )
+end )
+
+hook.Add( "EntityRemoved", "CFC_WeaponLockouts_LockWeapon", function( ent )
+    if not IsValid( ent ) or not ent:IsWeapon() then return end
+
+    CFCWeaponLockouts.lockWeapon( ent )
+end )
+
+hook.Add( "PlayerSwitchWeapon", "CFC_WeaponLockouts_TrackWeaponOwner", function( ply, old, new )
+    if not IsValid( new ) then return end
+
+    new.weaponLockoutOwner = ply
+end )
+
+hook.Add( "WeaponEquip", "CFC_WeaponLockouts_CanPickup", function( wep, ply )
+    if not IsValid( wep ) or not IsValid( ply ) then return end
+
+    local weaponClass = wep:GetClass()
+    ply.weaponLockoutWeapons = ply.weaponLockoutWeapons or {}
+    wep.weaponLockoutOwner = ply
+
+    local plyWeapons = ply.weaponLockoutWeapons
+    local isLocked = CFCWeaponLockouts.weaponIsLocked( ply, weaponClass )
+
+    --In certain cases, EntityRemoved gets called after WeaponEquip, causing odd behavior where the weapon is unlocked, meant to be locked, and not held by the player.
+    --Manually keeping track of the weapon classes held by a player allows us to catch that error.
+    if not isLocked and plyWeapons[weaponClass] and not CFCWeaponLockouts.NOT_LOCKABLE[wep:GetClass()] then
+        isLocked = true
+        CFCWeaponLockouts.lockWeapon( ply, wep, false )
+    end
+
+    plyWeapons[weaponClass] = true
+
+    if isLocked then
```

Comment:
How about you do

```lua
if not isLocked then
    lockWarns[ply][weaponClass] = "unlocked"
    lockWarns[ply].unlockedCount = ( lockWarns[ply].unlockedCount or 0 ) + 1

    return
end
```
Then put the rest of the code below it.

This'll let you dedent a huge section of this code 👍